inc/saitypes.h

```diff
@@ -989,6 +990,35 @@ typedef struct _sai_segment_list_t
     sai_ip6_t *list;
 } sai_segment_list_t;
 
+/**
+ * @brief JSON data type
+ * "attributes": [
+ * {
+ *    "attribute_name": {
+ *        "sai_metadata": {
+ *        "sai_attr_value_type": "<SAI_ATTR_VALUE_TYPE_T>",
+ *        "brief": "Brief Attribute Description",
+ *        "sai_attr_flags": "<SAI_ATTR_FLAGS_T>",
+ *        "allowed_object_types": [ "<LIST OF ALLOWED OBJECT TYPES>" ],
+ *        "default_value": "<DEFAULT ATTR VALUE>"
+ *        },
+ *        "value": <VALUE of the attribute>
+ *    }
+ * }
+ * ]
+ * attributes - Mandatory top-level key where JSON parsing begins
+ * attribute_name - Name of one attribute in the list of attributes
+ * sai_attr_value_type - Data type of the attribute
+ * brief - Optional description of the field
+ * sai_attr_flags - Optional Usage flags for the field
+ * allowed_object_types - If data type is OID, then this is the list of object types allowed as data
+ */
+typedef struct _sai_json_t
+{
+    /** String in JSON format */
+    sai_s8_list_t json;
+} sai_json_t;
```

Comment:
why we need special type if it's just wrapper ? and not use sai_s8_list_t directly?

Reply:
Having a new SAI_ATTR_VALUE_TYPE for json is very useful in generic code like sonic-sairedis. The JSON is allowed to carry other SAI datatypes like OID, IP address, etc and knowing that the attribute is a JSON helps with parsing it.
If I use straight, sai_s8_list_t, I don't think there is a way to distinguish between a simple string and a JSON string.
Would it help if we use a count/list within sai_json_t directly?

Reply:
no, lets leave this as is then